otelcol/collector.go

```diff
@@ -189,6 +178,12 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 
 	col.serviceConfig = &cfg.Service
 
+	conf := confmap.New()
+
+	if err = conf.Marshal(cfg); err != nil {
+		return fmt.Errorf("could not marshal configuration: %w", err)
```

Comment:
Another option could be to make this only a warning log, if we're comfortable that all extensions asking for the effective configuration will get an empty map.